src/vs/editor/contrib/multicursor/common/multicursor.ts

```diff
@@ -95,6 +95,19 @@ class InsertCursorAtEndOfEachLineSelected extends EditorAction {
 				});
 			}
 		}
-		editor.setSelections(newSelections);
+		return newSelections;
+	}
+
+	public run(accessor: ServicesAccessor, editor: ICommonCodeEditor): void {
+		let newSelections = new Array<ISelection>();
+		let selections = editor.getSelections();
+
+		for (var i = 0; i < selections.length; ++i) {
```

Comment:
This can be simplified to `let newSelections = selections.map((selection) => this.getCursorsForSelection(selection, editor));`